### DISCUSSION TITLE
how do I control when the conserver decides that a client is down?

### DISCUSSION THREAD

**Discussion by netllama** [bstansell/conserver#90] [Q&A]
TL;DR; how do I control when the conserver decides that a client is down?

I'm running conserver for several thousand (geographically) remote hosts which have OpenBMC for their serial console.  I'm doing this over an ssh tunnel in the console config (for each host).  For the most part, it works well as long as the console session is functional/responding on the remote host.

However, if a host's BMC stops working for some reason, the conserver almost never detects this behavior, and continues to claim that the console session is "up" (from "console -u" output).  This is obviously undesirable behavior, as it makes monitoring and other tasks painful.

The configuration that I'm using for each console session is as follows (hostnames and passwords obfuscated):

```
console                 c059.xyz001.ix.net  { type exec; master     127.0.0.1; host ipmi.net; exec sshpass -p 'XXX' ssh -e none -o "StrictHostKeyChecking=no" -o "ServerAliveInterval=60" -o "ServerAliveCountMax=2" -o "UserKnownHostsFile=/dev/null" -p 2200 slot2@ipmi.c059.xyz001.net; }
```

I was thinking of adding the ssh ConnectTimeout option to the command in the above config, but I'm not sure if that's going to actually influence the conserver's "down" detection in any way.  Currently, even when a client is truly down, and the ssh command is throwing errors, the conserver doesn't seem to treat the session as down much of the time.

Ideas/suggestions welcome

**Comment by netllama** [comment]
I found a live example of the problem.  In the conserver logs, I see the following while the client is 100% down (unreachable):
```

[Fri Jun  2 17:02:55 2023] conserver (5024): [c060.xyz001.ix.net.ix] automatic reinitialization
[Fri Jun  2 17:03:03 2023] conserver (5024): [c060.xyz001.ix.net.ix] exit(255)
[Fri Jun  2 17:03:03 2023] conserver (5024): [c060.xyz001.ix.net.ix] automatic reinitialization
[Fri Jun  2 17:03:10 2023] conserver (5024): [c060.xyz001.ix.net.ix] exit(255)
[Fri Jun  2 17:03:10 2023] conserver (5024): [c060.xyz001.ix.net.ix] automatic reinitialization
[Fri Jun  2 17:03:18 2023] conserver (5024): [c060.xyz001.ix.net.ix] exit(255)
[Fri Jun  2 17:03:18 2023] conserver (5024): [c060.xyz001.ix.net.ix] automatic reinitialization
[Fri Jun  2 17:03:25 2023] conserver (5024): [c060.xyz001.ix.net.ix] exit(255)
[Fri Jun  2 17:03:25 2023] conserver (5024): [c060.xyz001.ix.net.ix] automatic reinitialization
[Fri Jun  2 17:03:32 2023] conserver (5024): [c060.xyz001.ix.net.ix] exit(255)
[Fri Jun  2 17:03:32 2023] conserver (5024): [c060.xyz001.ix.net.ix] automatic reinitialization
[Fri Jun  2 17:03:40 2023] conserver (5024): [c060.xyz001.ix.net.ix] exit(255)
[Fri Jun  2 17:03:40 2023] conserver (5024): [c060.xyz001.ix.net.ix] automatic reinitialization
[Fri Jun  2 17:03:47 2023] conserver (5024): [c060.xyz001.ix.net.ix] exit(255)
[Fri Jun  2 17:03:47 2023] conserver (5024): [c060.xyz001.ix.net.ix] automatic reinitialization
[Fri Jun  2 17:03:55 2023] conserver (5024): [c060.xyz001.ix.net.ix] exit(255)
[Fri Jun  2 17:03:55 2023] conserver (5024): [c060.xyz001.ix.net.ix] automatic reinitialization
[Fri Jun  2 17:04:02 2023] conserver (5024): [c060.xyz001.ix.net.ix] exit(255)
[Fri Jun  2 17:04:02 2023] conserver (5024): [c060.xyz001.ix.net.ix] automatic reinitialization
[Fri Jun  2 17:04:10 2023] conserver (5024): [c060.xyz001.ix.net.ix] exit(255)
[Fri Jun  2 17:04:10 2023] conserver (5024): [c060.xyz001.ix.net.ix] automatic reinitialization
[Fri Jun  2 17:04:17 2023] conserver (5024): [c060.xyz001.ix.net.ix] exit(255)
[Fri Jun  2 17:04:17 2023] conserver (5024): [c060.xyz001.ix.net.ix] automatic reinitialization
```

Yet `console -u c060.xyz001.ix.net.ix` claims that its "up" the entire time.
